Makefile

```diff
@@ -0,0 +1,50 @@
+FEATURES_MINIMAL = --no-default-features --features 'std cargo' -p clap:3.0.0-beta.5
+FEATURES_FULL = --features 'wrap_help yaml regex unstable-replace unstable-multicall unstable-grouped'
+
+.PHONY: features-minimal features-full lint-minimal lint-full fmt-check lint coverage bench tests test debug clean
+
+features-minimal:
+	@echo "$(FEATURES_MINIMAL)"
+
+features-full:
+	@echo "$(FEATURES_FULL)"
+
+lint-minimal:
+	cargo clippy $(FEATURES_MINIMAL) -- -D warnings
+
+lint-full:
+	cargo clippy $(FEATURES_FULL) -- -D warnings
+
+fmt-check:
+	cargo fmt -- --check
+
+lint: lint-minimal lint-full fmt-check
+
+coverage:
+	cargo llvm-cov $(FEATURES_FULL) --lcov --output-path lcov.info
+
+bench:
+	cargo bench -- --output-format bencher
+
+tests:
+	cargo test $(FEATURES_FULL)
+
+test:
+ifeq (3, $(words $(MAKECMDGOALS)))
```

Comment:
Granted, my make experience was from an isolated set of developers but it seems like defining variables is more idiomatic than creating our own subcommand system

Reply:
`make test` and `make debug` are about running a single test with the test file and the test name being arguments. Since the arguments can change any time, we can't put them in a variable.

Reply:
People have `cargo`, I think its out of scope for us to be providing our own wrapper around it like this.

Reply:
Using `cargo` means adding all the features and then making sure to specify the arguments to the test file correctly. I think the wrapper here helps.

Reply:
How often does someone need to do that?  We previously already talked about that we didn't need a just file, this is just a different flavor of the same thing.

Reply:
As I said, `make debug` was pretty useful especially when testing the parsing logic bugs.

Reply:
Thats fine if you find it useful, you always have the option of keeping your own collection of scripts that help you.  They don't have to live in the repo.

Reply:
I will delete it but I think others will find `make debug` just as useful.

Reply:
I agree with @epage here. The common standard idiom here is to provide a variable in the makefile rather than programmatically checking how many arguments were provided... Best leave that to a shell script.

```
TEST_PKGS := clap_derive clap_up
```

Users can override those values with a `.env` file:

```
$ echo "TEST_PKGS=clap_derive" >> .env
$ source .env
$ make test
```

Then users who have differing opinions on what packages they want to test can maintain that `.env` file as part of their personal workflow.

The goal of a makefile is to provide a reproducible test environment. Personal scripts or makefiles should be held in a user's personal "helper utilities" repository, like a dotfiles or in a shared `contrib/` directory. But the goal of a "root" Makefile should be to provide a shared entrypoint to compile or test the project.

That being said... I'm curious why use a Makefile at all when cargo provides all of these features out of the box. You can tell what features cargo should test with `cargo test --all-features`, `cargo test --no-default-features`, and `cargo test` (with default features). Why define a FEATURES_FULL and FEATURES_MINIMAL when default features and `--all-features` should handle these cases in practice?